torch/distributions/beta.py

```diff
@@ -28,6 +28,7 @@ class Beta(Distribution):
     params = {'concentration1': constraints.positive, 'concentration0': constraints.positive}
     support = constraints.unit_interval
     has_rsample = True
+    mean_carrier_measure = float('nan')  # There is no closed form solution
```

Comment:
I think `NotImplementedError` would be more appropriate, and as you've implemented `ExponentialFamily`, simply omitting this attribute should correctly result in a `NotImplementedError`
```py
class ExponentialFamily(Distribution):
    @property
    def mean_carrier_measure(self):
        raise NotImplementedError
```

Reply:
Won't this break the entropy calculation? Should I add something like this?
```python
try:
    ...all entropy calculation code...
    result += self.mean_carrier_measure
except NotImplementedError:
    raise ValueError("There is no closed form solution for the entropy for non-constant carrier measure")
```

Reply:
I think I might have fixed it differently.

Reply:
I think the right behavior for `.entropy()` is to raise `NotImplementedError`, rather than return NANs. In particular, it's common in gradient estimators to e.g.
```py
try:
    entropy_estimate = dist.entropy()
except NotImplementedError:
    entropy_estimate = -dist_.log_prob(dist.sample(sample_shape=(1000,))).mean(-1)
```
If you're worried about the tests failing, I would simply wrap that line of the test in a try-except:
```
try:
    expected = dist.entropy()
except NotImplementedError:
    continue
```

Reply:
I think I fixed that. Please have a look at the latest push.

Reply:
You should really raise a `NotImplementedError` rather than a `ValueError`. The former means "the programmer has not implemented this; try another method", whereas the latter means "the user has input faulty data; fix your data".